main.tf

```
@@ -1,13 +1,54 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  tiered_caching  = local.argo_enabed && var.argo_tiered_caching_enabled ? "on" : "off"
+  smart_routing   = local.argo_enabed && var.argo_smart_routing_enabled ? "on" : "off"
+  argo_enabed     = module.this.enabled && var.argo_enabled
+  zone_enabled    = module.this.enabled && var.zone_enabled
+  zone_exists     = module.this.enabled && ! var.zone_enabled
+  records_enabled = module.this.enabled && length(var.records) > 0
+  zone_id         = local.zone_enabled ? join("", cloudflare_zone.default.*.id) : (local.zone_exists ? lookup(data.cloudflare_zones.default[0].zones[0], "id") : null)
```

`lookup` without a default is deprecated. 
```suggestion
  zone_id         = local.zone_enabled ? join("", cloudflare_zone.default.*.id) : (local.zone_exists ? data.cloudflare_zones.default[0].zones[0].id : null)
```